addons/edi_sale_forward_request/tests/test_sale_forward_request.py

```diff
@@ -52,9 +53,11 @@ def test01_basic(self):
         self.assertEqual(len(order_line), 1)
         self.assertEqual(order_line.product_uom_qty, 7.0)
 
-    def test01_invalid_json(self):
+    def test02_invalid_json(self):
         """Test json schema validation with invalid json"""
         doc = self.create_edi_document("order02.json")
-        doc.action_execute()
-        self.assertEqual(len(doc.issue_ids), 1)
-        self.assertTrue("Failed validating" in doc.issue_ids.name)
+        with self.assertRaisesIssue(doc, exception=ValidationError):
+            self.assertFalse(doc.action_execute())
```

Comment:
action_execute shouldn't throw errors

Reply:
No but `action_execute` will call `action_prepare` if the EDI doc isn't already prepared. I just kept it as `action_execute` as that what was in the test before but I can change this if preferred. :)

Reply:
action_prepare shouldn't throw an error either. Thats why we have issues.

Reply:
Theres a ```except Exception as err```  
https://github.com/unipartdigital/odoo-edi/blob/master/addons/edi/models/edi_document.py#L347

Reply:
Isn't that why we have `assertRaisesIssue` though? So it can mute issues that come up when loading the EDI.

For example here:
https://github.com/unipartdigital/odoo-edi/blob/master/addons/edi/tests/test_edi_document.py#L37

Reply:
Ah sorry I miss read that as ```assertRaises(```